File: Largest_Product_Grid/workout.py
from time import time

def itertativeSolution(grid):
	start = time()
	n = len(grid)
	maxProduct = 0
	for i in range(n):
		for j in range(n-3):
			row = grid[i][j]*grid[i][j+1]*grid[i][j+2]*grid[i][j+3]
			col = grid[j][i]*grid[j+1][i]*grid[j+2][i]*grid[j+3][i]
			maxProduct = max(maxProduct, row, col)
			if i < n-3:
				dg1 = grid[i][j]*grid[i+1][j+1]*grid[i+2][j+2]*grid[i+3][j+3]
				dg2 = grid[i][n-j-1]*grid[i+1][n-j-2]*grid[i+2][n-j-3]*grid[i+3][n-j-4]
				maxProduct = max(maxProduct, dg1, dg2)
	print("Using Loops, Ans : {}, Time : {} seconds".format(maxProduct, time()-start))

n = int(input("N : "))
grid = [list(map(int,input().split())) for _ in range(n)]

itertativeSolution(grid)

File: Maximum_Path_Sum_I/workout.js
const fs = require('fs');
const {performance} = require('perf_hooks');

let input = fs.readFileSync('./input.txt', 'utf8');
input = input.replace(/(\r\n|\n|\r)/gm, ",").split(',');

let n = +input[0];
let triangle = [];
for (let i = 1; i <= n; i += 1)
	triangle.push(input[i].split(' ').map(Number));

const directSolution = (size, arr) => {
	const start = performance.now();
	let row = size - 2;
	let arrCpy = arr.slice();
	while (row >= 0){
		for (let i = 0; i < arrCpy[row].length; i += 1)
			arrCpy[row][i] += Math.max(arrCpy[row+1][i], arrCpy[row+1][i+1]);
		row -= 1;
	}
	const answer = arrCpy[0][0];
	console.log(`Direct Solution, Ans : ${answer}, Time : ${(performance.now()-start)/1000} seconds`);
};

directSolution(n, triangle);